File: src/API/src/Modules/Leads/Leads.Application/DTOs/CreateLeadRequest.cs
namespace Leads.Application.DTOs
{
    public class CreateLeadRequest
    {
        public Guid SourceId { get; set; }
        public Guid CategoryId { get; set; }
        public Guid? PartnerId { get; set; }
        public string? CampaignName { get; set; }
        public string CompanyName { get; set; } = string.Empty;
        public string? Website { get; set; }
        public string ContactPersonName { get; set; } = string.Empty;
        public string? JobTitle { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? AlternatePhone { get; set; }
        public Guid CountryId { get; set; }
        public string? Address { get; set; }
        public Guid? IndustryId { get; set; }
        public string? Notes { get; set; }
        public int? LeadScore { get; set; }
        public List<Guid> ProductIds { get; set; } = new();
    }
}

File: src/ClientApp/src/app/pages/leads/dtos/create-lead.request.ts
export interface CreateLeadRequest {
    sourceId: string;
    categoryId: string;
    partnerId?: string | null;
    campaignName?: string | null;
    sourceStartDate?: string | null;
    sourceEndDate?: string | null;
    companyName: string;
    website?: string | null;
    contactPersonName: string;
    jobTitle?: string | null;
    email?: string | null;
    phone?: string | null;
    alternatePhone?: string | null;
    countryId: string;
    address?: string | null;
    industryId?: string | null;
    notes?: string | null;
    leadScore?: number | null;
    productIds: string[];
}
